### DISCUSSION TITLE
ISR SSG page gives "404 Not Found", but only in Vercel

### DISCUSSION THREAD

**Discussion by fullyherge** [vercel/community#5935] [Help]
### Summary

I'm at the end of my rope with this now...

A specific page in my application suddenly started responding with "404 Not Found" **but only when deployed to Vercel**. Running locally, either via `npm run dev` or `npm run start` with a production build both work perfectly fine. The output in the build log is, as far as I can tell, identical between Vercel and local build.

Anyone have any clues where I can figure this out? All google results give the same "use rewrites to redirect everything back to /" which doesn't seem applicable in a Next 14 App Router application scenario.

### Steps to Reproduce

**Log from `generateStaticParams` in Vercel**

![Screenshot 2024-02-27 at 13 10 38](https://github.com/vercel/community/assets/134703250/385fc330-e854-4a40-8058-898e6f1d5607)

**Further down the log, in the build Route summary**

![Screenshot 2024-02-27 at 13 12 27](https://github.com/vercel/community/assets/134703250/9a32faf5-2678-4fa6-b535-0bb8936868aa)

**Vercel deployment ISR summary, missing Career**

![Screenshot 2024-02-27 at 13 15 27](https://github.com/vercel/community/assets/134703250/7917e112-6fd6-478b-9d3d-c1039166bb2d)

**404 Network request for deployed version**

![Screenshot 2024-02-27 at 13 13 16](https://github.com/vercel/community/assets/134703250/c64a246b-269c-4153-bad4-0d9c47648ca4)

**Some context and questionsmarks**
- The route is printed in the Route output of the build step, but its' corresponding ISR function is not shown in the output in Vercel?
- It's one of 10 dynamic pages in `src/(site)/[...slug]/page.tsx`
![Screenshot 2024-02-27 at 13 36 06](https://github.com/vercel/community/assets/134703250/e71b5913-bab4-44ff-a99d-7b37ee680ba0)

- Removing one specific Server Component block in the CMS on this particular page makes the page found again...? But if this component has some sort of issue it's not showing any errors anywhere, and I don't see why it would work locally without issue then?

I also tried doing `vercel build` and then `vercel deploy` from my terminal to make sure it's not the Github connection acting up somehow, but no luck.

```
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.2.0: Wed Nov 15 21:55:06 PST 2023; root:xnu-10002.61.3~2/RELEASE_ARM64_T6020
Binaries:
  Node: 18.18.2
  npm: 9.8.1
  Yarn: 1.22.21
  pnpm: 8.9.2
Relevant Packages:
  next: 14.1.0 (also tested with 14.1.1-canary.74)
  eslint-config-next: 14.1.0
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.2.2
Next.js Config:
  output: N/A
```

**Comment by fullyherge** [comment]
> Removing one specific Server Component block in the CMS on this particular page makes the page found again...? But if this component has some sort of issue it's not showing any errors anywhere, and I don't see why it would work locally without issue then?

In this component I had a fetch, written like this:

```tsx
    const response = await fetch(`${endpoint}?format=json`, {
      cache: "no-cache", // also tried "no-store" with same issue
    });
    const jobData = await response.json();
    console.log("jobs fetched", jobData.length);
```

I tried, on a whim, to change this to:

```tsx
    const response = await fetch(`${endpoint}?format=json`, {
      next: {
        revalidate: 180,
      },
    });
    const jobData = await response.json();
    console.log("jobs fetched", jobData.length);
```

which for some unapparent reason seems to be working?... I feel like there's a bug in Vercel here.

**Comment by fullyherge** [comment]
Posted an issue in Next repo [here](https://github.com/vercel/next.js/issues/62634)

**Comment by kasvith** [comment]
Hey, im facing the same issue but with pages router

It wasnt happening before, now often i see some pages are present in the deployment, but they gives 404
we had `fallback: "blocking"` which worked without a problem

recently i saw even static pages giving 404

**Comment by leerob** [comment]
Thanks for sharing 🙏 We'll take a look into this.

**Comment by ijjk** [comment]
Hi, a fix for this has been rolled out, please re-deploy and give it a try!

[a comment by an automated account is not shown]
